proto/minder/v1/minder.proto

```diff
@@ -1944,19 +1981,43 @@ message ArtifactType {
 // GitType defines the git data ingester.
 message GitType {
     // clone_url is the url of the git repository.
-    string clone_url = 1;
+    string clone_url = 1 [
+        (buf.validate.field).string = {
+            uri: true,
+            max_len: 200,
+        },
+        (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
+    ];
 
     // branch is the branch of the git repository.
-    string branch = 2;
+    string branch = 2 [
+        (buf.validate.field).string = {
+            pattern: "^[[:word:]./-]+$",
+            max_len: 200,
+        },
+        (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
+    ];
 }
 
 // DiffType defines the diff data ingester.
 message DiffType {
     message Ecosystem {
         // name is the name of the ecosystem.
-        string name = 1;
+        string name = 1 [
+            (buf.validate.field).string = {
+                pattern: "^[a-z]+(_[a-z]+)*$",
+                min_len: 1,
+                max_len: 200,
+            }
+        ];
         // depfile is the file that contains the dependencies for this ecosystem
-        string depfile = 2;
+        string depfile = 2 [
+            (buf.validate.field).string = {
+                pattern: "^(\\./)?([a-zA-Z0-9_\\-]+/)*[a-zA-Z0-9_\\-]+(\\.[a-zA-Z0-9]+)?$",
```

Comment:
We don't want to allow `.` in the filename for the depfile path or multiple dots in the requirements file?  (The python "standard" for development dependencies is `requirements-dep.txt`, but given how loose that standard is, I could see `requirements.dep.txt` slipping in.)

Reply:
yeah, I wasn't sure about this too, but I'm okay to leave it for now because of the reasons you mentioned 👍